src/colorizer/ColorizeCanvas2D.ts

```diff
@@ -714,11 +724,40 @@ export default class ColorizeCanvas2D implements IRenderCanvas {
     return frameLoadResult;
   }
 
+  public getScreenSpaceMatrix(): Matrix4 {
+    if (!this.params || !this.params.dataset) {
+      return new Matrix4();
+    }
+    // 1. Go from centroid coordinates (in frame pixels) to normalized frame
+    //    coordinates, where (0,0) is the center of the frame and axes are in
+    //    the [-0.5, 0.5] range.
+    const frameResolution = this.params.dataset.frameResolution;
+    const framePixelsToNormFrameCoords = new Matrix4().compose(
+      new Vector3(-0.5, -0.5, 0), // Shift so (0,0) is the center of the frame
+      new Quaternion(), // No rotation
+      new Vector3(1 / frameResolution.x, 1 / frameResolution.y, 1) // Scale to normalized coordinates
+    );
+
+    // 2. Apply pan offset, flipping Y axis.
+    const panningOffset = new Matrix4().makeTranslation(this.panOffset.x, this.panOffset.y * -1, 0);
+
+    // 3. Scale back to onscreen canvas pixels, and move origin back to top left corner.
+    const frameToCanvasPxScale = this.scaleInfo.frameToCanvasCoordinates.clone().multiply(this.canvasResolution);
+    const normFrameCoordsToCanvasPixels = new Matrix4().compose(
+      new Vector3(...this.canvasResolution.clone().multiplyScalar(0.5).toArray(), 0), // Move origin to top left corner
+      new Quaternion(), // No rotation
+      new Vector3(frameToCanvasPxScale.x, frameToCanvasPxScale.y, 0) // Scale to canvas pixels, also Z=0
+    );
+
+    // Combine all transformations into a single matrix
+    return normFrameCoordsToCanvasPixels.multiply(panningOffset).multiply(framePixelsToNormFrameCoords);
```

Comment:
nit: there is tons of room to optimize this function.  is this only called at most once per rendered frame?

Reply:
Yes! It's called just the one time to setup for the overlay canvas' render.

Reply:
@meganrm I'm not sure that it does since this logic isn't reused between the 2D and 3D canvases. The only thing I could see being useful would be to name what each step is doing explicitly, but they'd each only be returning a single method call (`Matrix4().compose(...)`)